src/pages/Script/index.tsx

```diff
@@ -0,0 +1,314 @@
+  capacityOfReferringCells: '',
+}
+
+const initScriptPageInfo: ScriptPageInfo<any> = {
+  total: 0,
+  data: [],
+  status: 'None',
+  page: 1,
+  size: 10,
+}
+
+export const updatePageSection = (
+  fetchFunc: () => Promise<Response.Response<Response.Wrapper<any[]>> | null>,
+  setScriptInfo: Dispatch<SetStateAction<ScriptInfo>>,
+  setPagesInfo: Dispatch<SetStateAction<ScriptPageInfo<any>>>,
+  pagesInfo: ScriptPageInfo<any>,
+  pageDataName: 'ckbTransactions' | 'deployedCells' | 'referringCells',
+) => {
+  setPagesInfo({
+    ...pagesInfo,
+    status: 'InProgress',
+  })
+  fetchFunc()
+    .then(response => {
+      const data = response!.data as any
+      const meta = response!.meta as Response.Meta
+      setScriptInfo({
+        scriptName: data.scriptName,
+        scriptType: data.scriptType,
+        typeId: data.TypeId,
+        codeHash: data.codeHash,
+        hashType: data.hashType,
+        capacityOfDeployedCells: data.capacityOfDeployedCells,
+        capacityOfReferringCells: data.capacityOfReferringCells,
+      })
+      setPagesInfo({
+        ...pagesInfo,
+        data: data[pageDataName],
+        total: meta ? meta.total : 0,
+        status: 'OK',
+      })
+    })
+    .catch(error => {
+      const isEmpty = error && error.response && error.response.status === 404
+      setPagesInfo({
+        ...pagesInfo,
+        data: [],
+        total: 0,
+        status: isEmpty ? 'OK' : 'Error',
+      })
+    })
+}
+
+export const ScriptPage = () => {
+  const history = useHistory()
+  const { search, hash } = useLocation()
+  const [activeKey, setActiveKey] = useState<'transactions' | 'deployed_cells' | 'referring_cells'>('transactions')
+  const { codeHash } = useParams<{ codeHash: string }>()
+  const { hashType } = useParams<{ hashType: string }>()
+  const parsed = queryString.parse(search)
+  const [scriptInfo, setScriptInfo] = useState<ScriptInfo>(initScriptInfo)
+  const [ckbTransaction, setCkbTransaction] = useState<ScriptPageInfo<any>>(initScriptPageInfo)
+  const [deployedCell, setDeployedCell] = useState<ScriptPageInfo<any>>(initScriptPageInfo)
+  const [referringCell, setReferringCell] = useState<ScriptPageInfo<any>>(initScriptPageInfo)
+
+  const currentPage = parsePageNumber(parsed.page, PageParams.PageNo)
+  const pageSize = parsePageNumber(parsed.size, PageParams.PageSize)
+
+  useEffect(() => {
+    const hashStart = hash.indexOf('#')
+    const pageInfo = {
+      ...initScriptPageInfo,
+      page: currentPage,
+      size: pageSize,
+    }
+    if (hashStart < 0) {
+      setActiveKey('transactions')
+      updatePageSection(
+        () => fetchScriptsCkbTransactionsInfo(codeHash, hashType, currentPage, pageSize),
+        setScriptInfo,
+        setCkbTransaction,
+        pageInfo,
+        'ckbTransactions',
+      )
+    } else {
+      const hashEnd = hash.indexOf('?')
+      const anchor = hashEnd < hashStart ? hash.substring(hashStart + 1) : hash.substring(hashStart + 1, hashEnd)
```

Comment:
`tab` could be a query field in `search`, or specified by `params` of react-router as `useParams`(https://v5.reactrouter.com/web/api/Hooks/useparams)